GRDB/Record/PersistableRecord+Encodable.swift

```diff
@@ -169,7 +185,7 @@ private struct PersistableRecordEncoder : Encoder {
     func container<Key>(keyedBy type: Key.Type) -> KeyedEncodingContainer<Key> {
         // Asked for a keyed type: top level required
         guard codingPath.isEmpty else {
-            fatalError("unkeyed encoding is not supported")
+            return KeyedEncodingContainer(ThrowingKeyedContainer(error: EncodingError.invalidValue(codingPath.isEmpty, EncodingError.Context(codingPath: codingPath, debugDescription: "unkeyed encoding is not supported"))))
```

Comment:
I think we should throw an error. This error will be caught during "natural" encoding, and trigger JSON encoding.

Ideally, this error should be an EncodingError.

And it will allow us to apply my comment on https://github.com/groue/GRDB.swift/pull/397/files#diff-e9cae7906e97e4aadddcbc90115862c3R87.

Eventually we discover that we don't need ThrowingKeyedContainer.

Reply:
If we change func container<Key>(keyedBy type: Key.Type) -> KeyedEncodingContainer<Key>  to throw we lose conformity to the Encoder protocol

Reply:
Ha, I understand: `container(keyedBy:)` does not throw. Thanks for fixing my mistake.